### PR TITLE
Use `chembl_downloader` for generating substructure library

### PR DESCRIPTION
**Pull request by cthoyt**
The [`chembl_downloader`](https://github.com/cthoyt/chembl-downloader) automates downloading the ChEMBL SDF dump, and there's a little code that wraps gunzipping it and creating a ForwardSDMolSupplier in a context generator. This will hopefully make it a bit easier for someone else to use this code!

The updated notebook can be viewed on the NBViewer app at https://nbviewer.jupyter.org/github/cthoyt/rdkit_blog/blob/update-notebook/notebooks/Generalized%20Substructure%20Search.ipynb.

I wrote more about it on my blog https://cthoyt.com/2021/08/04/taming-chembl-sdf.html - this package could be the subject of a lightning talk at a future RDKit UGM for enabling more reproducible use of ChEMBL

### CODE REVIEW

**Comment by cthoyt** [comment]
Hi @greglandrum, any chance you will consider this?

File: notebooks/Generalized Substructure Search.ipynb [hunk 1]
```diff
@@ -75,20 +75,20 @@
...
-    "data = []\n",
-    "for i,mol in enumerate(suppl):\n",
-    "    if not ((i+1)%50000):\n",
-    "        print(f\"Processed {i+1} molecules in {(time.time()-t1):.1f} seconds\")\n",
-    "    if mol is None or mol.GetNumAtoms()>50:\n",
-    "        continue\n",
-    "    fp = Chem.PatternFingerprint(mol,fpSize=1024,tautomerFingerprints=True)\n",
-    "    smi = Chem.MolToSmiles(mol)\n",
-    "    data.append((smi,fp))\n",
+    "with chembl_downloader.supplier(version=\"29\") as suppl:\n",
```
Comment (greglandrum) [inline notebooks/Generalized Substructure Search.ipynb:81]: I'd suggest modifying adding a comment above this mentioning that it's using your chembl_downloader and linking to the github repo and possibly your blog post.

**Review by greglandrum — CHANGES_REQUESTED** [review]
Thanks for this!
The changes look good, but I think it would be better to include a comment with a link to your downloader.

File: notebooks/Generalized Substructure Search.ipynb (hunk 1 above)
Comment (cthoyt) [reply notebooks/Generalized Substructure Search.ipynb:81]: Sorry for taking so long, but I have added the requested additional text in 09befa2. Cheers!

**Review by greglandrum — APPROVED** [review]
Thanks @cthoyt !

**Pull request merged**